x-pack/plugins/monitoring/server/alerts/ccr_read_exceptions_alert.ts

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { BaseAlert } from './base_alert';
+import {
+  AlertData,
+  AlertCluster,
+  AlertState,
+  AlertMessage,
+  CCRReadExceptionsUIMeta,
+  AlertMessageTimeToken,
+  AlertMessageLinkToken,
+  AlertInstanceState,
+  CommonAlertParams,
+} from '../../common/types/alerts';
+import { AlertInstance } from '../../../alerts/server';
+import {
+  INDEX_PATTERN_ELASTICSEARCH,
+  ALERT_CCR_READ_EXCEPTIONS,
+  ALERT_DETAILS,
+} from '../../common/constants';
+import { fetchCCRReadExceptions } from '../lib/alerts/fetch_ccr_read_exceptions';
+import { getCcsIndexPattern } from '../lib/alerts/get_ccs_index_pattern';
+import { AlertMessageTokenType, AlertSeverity } from '../../common/enums';
+import { SanitizedAlert } from '../../../alerts/common';
+import { AlertingDefaults, createLink } from './alert_helpers';
+import { appendMetricbeatIndex } from '../lib/alerts/append_mb_index';
+import { Globals } from '../static_globals';
+
+export class CCRReadExceptionsAlert extends BaseAlert {
+  constructor(public rawAlert?: SanitizedAlert) {
+    super(rawAlert, {
+      id: ALERT_CCR_READ_EXCEPTIONS,
+      name: ALERT_DETAILS[ALERT_CCR_READ_EXCEPTIONS].label,
+      throttle: '6h',
+      defaultParams: {
+        duration: '1h',
+      },
+      actionVariables: [
+        {
+          name: 'remoteClusters',
+          description: i18n.translate(
+            'xpack.monitoring.alerts.ccrReadExceptions.actionVariables.remoteClusters',
+            {
+              defaultMessage: 'List of remote clusters that are experiencing CCR read exceptions.',
+            }
+          ),
+        },
+        {
+          name: 'followerIndices',
+          description: i18n.translate(
+            'xpack.monitoring.alerts.ccrReadExceptions.actionVariables.followerIndices',
+            {
+              defaultMessage: 'List of follower indices reporting CCR read exceptions.',
+            }
+          ),
+        },
+        ...Object.values(AlertingDefaults.ALERT_TYPE.context),
+      ],
+    });
+  }
+
+  protected async fetchData(
+    params: CommonAlertParams,
+    callCluster: any,
+    clusters: AlertCluster[],
+    availableCcs: string[]
+  ): Promise<AlertData[]> {
+    let esIndexPattern = appendMetricbeatIndex(Globals.app.config, INDEX_PATTERN_ELASTICSEARCH);
+    if (availableCcs) {
+      esIndexPattern = getCcsIndexPattern(esIndexPattern, availableCcs);
+    }
+    const { duration } = params;
```

Comment:
I can't seem to find the conversation, but I'm recalling we agreed to leverage `parseDuration` to convert the string-like date to a number, so we don't need to assume the syntax will match the date math syntax from ES. Here is an example: https://github.com/elastic/kibana/blob/master/x-pack/plugins/monitoring/server/alerts/cpu_usage_alert.ts#L78

I'm going to ping @mikecote to see if he remembers this conversation

Reply:
I can't recall much from that conversation but I think the `parseDuration` function didn't support all the possible values.